content/en/blog/_posts/2021-09-06-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: blog
+title: 'Alpha in Kubernetes v1.22: Quality-of-Service for Memory Resources'
+date: 2021-09-01
+slug: qos-memory-resources
+---
+
+**Authors:** Tim Xu (Tencent Cloud)]
+
+Kubernetes v1.22, released in August 2021, introduced a new alpha feature that improves how Linux nodes implement memory resource requests and limits.
+
+In prior releases, Kubernetes did not support memory quality guarantees. 
+For example, if you set container resources as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+  - name: nginx
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "64Mi"
+        cpu: "500m"
+```
+`spec.containers[].resources.requests`(e.g. cpu, memory) is designed for scheduling. When you create a Pod, the Kubernetes scheduler selects a node for the Pod to run on. Each node has a maximum capacity for each of the resource types: the amount of CPU and memory it can provide for Pods. The scheduler ensures that, for each resource type, the sum of the resource requests of the scheduled Containers is less than the capacity of the node.
+
+`spec.containers[].resources.limits` is passed to the container runtime when the kubelet starts a container. CPU is considered a "compressible" resource. If your app starts hitting your CPU limits, Kubernetes starts throttling your container, giving your app potentially worse performance. However, it won’t be terminated. That is what "compressible" means.
```

Comment:
I would add a little paragraph below with something like this, what do you think? I think helps it make it a bit more clear the impact of this change :) 


> In cgroupv1, and prior to this feature, the container runtime never took into account and effectively ignored `spec.containers[].resources.requests["memory"]`. This is unlike CPU, in which the container runtime consider both requests and limits.